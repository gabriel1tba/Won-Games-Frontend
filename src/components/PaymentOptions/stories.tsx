import { Story, Meta } from '@storybook/react/types-6-0';
import PaymentOptions, { PaymentOptionsProps } from '.';

import cardsMock from './mock';

export default {
  title: 'PaymentOptions',
  component: PaymentOptions,
  args: {
    cards: cardsMock,
  },
} as Meta;

export const Default: Story<PaymentOptionsProps> = (args) => (
  <PaymentOptions {...args} />
);
