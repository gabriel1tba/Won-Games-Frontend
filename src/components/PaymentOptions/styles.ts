import styled from 'styled-components';

export const Wrapper = styled.div``;

export const Body = styled.div``;

export const Footer = styled.div``;

export const CardsList = styled.div``;

export const CardItem = styled.div``;

export const CardInfo = styled.div``;

export const AddCard = styled.div``;
