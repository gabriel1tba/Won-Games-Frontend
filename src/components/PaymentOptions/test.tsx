import { screen } from '@testing-library/react';
import renderWithTheme from 'utils/tests/helpers';

import PaymentOptions from '.';

import cardsMock from './mock';

describe('<PaymentOptions />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(
      <PaymentOptions
        handlePayment={() => {
          console.log('test');
        }}
        cards={cardsMock}
      />,
    );

    expect(
      screen.getByRole('heading', { name: /Payment/i }),
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
