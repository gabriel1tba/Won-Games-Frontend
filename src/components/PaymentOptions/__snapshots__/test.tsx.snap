// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentOptions /> should render the heading 1`] = `
.c3 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
  width: 100%;
  background: none;
  color: #F231A5;
}

.c4:hover {
  background: none;
}

.c4:hover {
  color: #e20e8d;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
  width: 100%;
}

.c5:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c5 svg {
  width: 1.5rem;
}

.c5 svg + span {
  margin-left: 0.8rem;
}

.c0 {
  color: #030517;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 1.6rem;
}

.c0::after {
  position: absolute;
  left: 0;
  bottom: -1rem;
  content: '';
  width: 5rem;
  border-bottom: 0.5rem solid #F231A5;
}

.c0::after {
  width: 3rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 1.8rem;
  height: 1.8rem;
  border: 0.2rem solid #F231A5;
  border-radius: 50%;
  background: transparent;
  -webkit-transition: background 0.1s ease-in-out;
  transition: background 0.1s ease-in-out;
  outline: none;
  cursor: pointer;
}

.c2:focus {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c2:before {
  content: '';
  width: 0.8rem;
  height: 0.8rem;
  border-radius: 50%;
  background: #F231A5;
  -webkit-transition: opacity 0.1s ease-in-out;
  transition: opacity 0.1s ease-in-out;
  opacity: 0;
  position: absolute;
}

.c2:checked:before {
  opacity: 1;
}

<div
  class=""
>
  <div
    class=""
  >
    <h2
      class="c0"
      color="black"
    >
      Payment
    </h2>
    <div
      class=""
    >
      <div
        class=""
      >
        <div
          class=""
        >
          <img
            alt="visa"
            src="/img/visa.png"
          />
          *** *** **** 4325
        </div>
        <div
          class="c1"
        >
          <input
            class="c2"
            id="*** *** **** 4325"
            name="credit-card"
            type="radio"
            value="*** *** **** 4325"
          />
        </div>
      </div>
      <div
        class=""
      >
        <div
          class=""
        >
          <img
            alt="mastercard"
            src="/img/master-card.png"
          />
          *** *** **** 4326
        </div>
        <div
          class="c1"
        >
          <input
            class="c2"
            id="*** *** **** 4326"
            name="credit-card"
            type="radio"
            value="*** *** **** 4326"
          />
        </div>
      </div>
      <div
        class=""
        role="button"
      >
        <svg
          aria-hidden="true"
          class="c3"
          fill="currentColor"
          focusable="false"
          height="14"
          viewBox="0 0 24 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
         Add a new credit card
      </div>
    </div>
  </div>
  <div
    class=""
  >
    <a
      class="c4"
    >
      <span>
        Continue shopping
      </span>
    </a>
    <button
      class="c5"
    >
      <svg
        aria-hidden="true"
        class="c3"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49A.996.996 0 0020.01 4H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
      <span>
        Buy now
      </span>
    </button>
  </div>
</div>
`;
